waiting/__init__.py

```diff
@@ -13,8 +13,9 @@
 
 def wait(*args, **kwargs):
     result = _Result()
-    for x in iterwait(result=result, *args, **kwargs):
-        pass
+    for exc in iterwait(result=result, *args, **kwargs):
+        if exc:
+            raise exc
```

Comment:
Also, if for some reason we were to add an actual value to iterwait indicating the state (timeout, number of attempts etc) this would break...